src/mono/wasm/features.md

```diff
@@ -2,36 +2,311 @@
+
+## Content security policy
+
+dotnet runtime for wasm is CSP compliant starting from .Net 8, except legacy JS interop methods.
+
+In order to enable it, please set HTTP headers similar to `Content-Security-Policy: default-src 'self' 'wasm-unsafe-eval'`
+
+HTTP headers are sent by your HTTP server or proxy, which need to be properly configured.
+
+See also [CSP on MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP)
+
+See also [wasm-unsafe-eval](https://github.com/WebAssembly/content-security-policy/blob/main/proposals/CSP.md#the-wasm-unsafe-eval-source-directive)
+
+## Globalization, ICU
+
+Browsers don't offer full set APIs for working with localization and so we have to bring the data and the logic as part of the application.
+
+In order to make downloades smaller, runtime will detect the locale of the browser and load just shard of the ICU database. You can also create and configure your own custom shards.
+
+For more details see [globalization-icu-wasm.md](../../../docs/design/features/globalization-icu-wasm.md)
+
+If your application doesn't need to work with locatization, you can reduce download size by `<InvariantCulture>true</InvariantCulture>`.
+
+For more details see [globalization-invariant-mode.md](../../../docs/design/features/globalization-invariant-mode.md)
+
+If your application needs to work with locatization but you don't require processing speed for working with large texts, you can reduce download size by `<HybridGlobalization>true</HybridGlobalization>`.
+It will use browser APIs instead part of ICU database. This feature is still in development.
+
+For more details see [globalization-hybrid-mode.md](../../../docs/design/features/globalization-hybrid-mode.md)
+
+This requires that you have [wasm-tools workload](#wasm-tools-workload) installed.
+
+## Timezones
+
+Browsers don't offer API for working with time zone database and so we have to bring the time zone data as part of the application.
+
+If your application doesn't need to work with time zones, you can reduce download size by `<InvariantTimezone>true</InvariantTimezone>`.
+
+This requires that you have [wasm-tools workload](#wasm-tools-workload) installed.
+
+## Bundling JavaScript and other assets
+
+In the web ecosystem it's usual that developers use assets bundlers like [webpack](https://github.com/webpack/webpack) or [rollup](https://github.com/rollup/rollup) to bundle many files into one large .js file.
+
+You can bundle the `dotnet.js` ES6 module with the rest of your JavaScript application.
+
+The other assets and JS modules of the dotnet are loaded via dynamic `import` or via `fetch` APIs. They are not ready to be bundled in Net8.
+We consider that the dotnet application is usually large as is and giving the browser chance to start compiling it in parallel with other downloads is better.
+
+We would like to [hear from the community](https://github.com/dotnet/runtime/issues/86162) more about the use-cases when it would be benefitial.
```

Comment:
```suggestion
We would like to [hear from the community](https://github.com/dotnet/runtime/issues/86162) more about the use-cases where it would be beneficial.
```